pkg/sidecar/sidecar.go

```diff
@@ -183,6 +189,106 @@ func (s *Sidecar) dumpBundles(svidResponse *workloadapi.X509Context) error {
 	return nil
 }
 
+func (s *Sidecar) writeJSON(fileName string, certs map[string]interface{}) {
+	file, err := json.Marshal(certs)
+	if err != nil {
+		s.config.Log.Errorf("Unable to parse certs: %v", err)
+	}
+
+	jsonPath := path.Join(s.config.CertDir, fileName)
+	err = os.WriteFile(jsonPath, file, os.ModePerm)
+	if err != nil {
+		s.config.Log.Errorf("Unable to write JSON file: %v", err)
+	}
+}
+
+func (s *Sidecar) updateJWTBundle(jwkSet *jwtbundle.Set) {
+	s.config.Log.Info("Updating JWT bundle")
+
+	bundles := make(map[string]interface{})
+	for _, bundle := range jwkSet.Bundles() {
+		bytes, err := bundle.Marshal()
+		if err != nil {
+			s.config.Log.Errorf("Unable to marshal JWT bundle: %v", err)
+			continue
+		}
+		bundles[bundle.TrustDomain().Name()] = base64.StdEncoding.EncodeToString(bytes)
+	}
+
+	s.writeJSON(s.config.JWTBundleFilename, bundles)
+}
+
+func (s *Sidecar) fetchJWTSVID(options ...workloadapi.ClientOption) (*jwtsvid.SVID, error) {
+	clientOptions := workloadapi.WithClientOptions(options...)
+
+	jwtSource, err := workloadapi.NewJWTSource(context.Background(), clientOptions)
+	if err != nil {
+		s.config.Log.Errorf("Unable to create JWTSource: %v", err)
+		return nil, err
+	}
+	defer jwtSource.Close()
+
+	jwtSVID, err := jwtSource.FetchJWTSVID(context.Background(), jwtsvid.Params{Audience: s.config.JWTAudience})
+	if err != nil {
+		s.config.Log.Errorf("Unable to fetch JWT SVID: %v", err)
+		return nil, err
+	}
+
+	_, err = jwtsvid.ParseAndValidate(jwtSVID.Marshal(), jwtSource, []string{s.config.JWTAudience})
+	if err != nil {
+		s.config.Log.Errorf("Unable to parse or validate token: %v", err)
+		return nil, err
+	}
+
+	return jwtSVID, nil
+}
+
+func getRetryInterval() func() time.Duration {
+	backoffInterval := 0
+	return func() time.Duration {
+		backoffInterval += 5
+		return time.Duration(backoffInterval) * time.Second
+	}
+}
+
+func getRefreshInterval(svid *jwtsvid.SVID) time.Duration {
+	return time.Until(svid.Expiry)/2 + time.Second
+}
+
+func (s *Sidecar) updateJWTSVID(ctx context.Context, options ...workloadapi.ClientOption) {
+	retryInterval := getRetryInterval()
+
+	ticker := time.NewTicker(time.Second)
```

Comment:
shouldn't the ticker be set to the `refreshInterval` (using `getRefreshInterval`) ?

Reply:
I think it's a possibility too, but if its the first time where an JWT SVID is going to be fetched I don't see why the helper should wait.

Reply:
Your implementation introduces a one-second delay before the first SVID update, which seems unnecessary. It might be more efficient to execute the first SVID fetch immediately and then enter the loop with a ticker set to an interval based on token expiry. This approach would benefit from extracting the SVID fetching logic into a separate function, as shown below:

```
func (s *Sidecar) updateJWTSVID(ctx context.Context, options ...workloadapi.ClientOption) {
	// Immediate first update
	s.performJWTSVIDUpdate(options...)

	// Set ticker for subsequent updates
	ticker := time.NewTicker(getRefreshInterval())
	defer ticker.Stop()

	for {
		select {
		case <-ctx.Done():
			return
		case <-ticker.C:
			s.performJWTSVIDUpdate(options...)
			ticker.Reset(getRefreshInterval())
		}
	}
}
```

This change ensures immediate responsiveness and more logical timing for updates.